pipelines/google_analytics/__init__.py

```diff
@@ -0,0 +1,163 @@
+                     property_id: int = dlt.config.value,
+                     rows_per_page: int = dlt.config.value,
+                     queries: List[DictStrAny] = dlt.config.value,
+                     start_date: str = dlt.config.value,
+                     end_date: str = dlt.config.value) -> List[DltResource]:
+    """
+    The DLT source for Google Analytics. Will load basic Analytics info to the pipeline.
+    :param credentials: Credentials to the Google Analytics Account
+    :param scope: The scope of oauth token permissions, must match the scope of the refresh tokens.
+    :param redirect_uri: The redirect uri specified in the oauth client.
+    :param property_id: A reference to the Google Analytics project. https://developers.google.com/analytics/devguides/reporting/data/v1/property-id
+    :param rows_per_page: Controls how many rows are retrieved per page in the reports. Default is 10000, maximum possible is 100000.
+    :param queries: List containing info on the all the reports being requested with all the dimensions and metrics per report.
+    :param start_date: Needs to be the string version of date in the format yyyy-mm-dd and some other values: https://developers.google.com/analytics/devguides/reporting/data/v1/rest/v1beta/DateRange
+    can be left empty for default incremental load behaviour.
+    :param end_date: Needs to be the string version of date in the format yyyy-mm-dd and some other values: https://developers.google.com/analytics/devguides/reporting/data/v1/rest/v1beta/DateRange
+    can be left empty for default incremental load behaviour.
+    :return resource_list: list containing all the resources in the Google Analytics Pipeline.
+    """
+
+    # generate access token for credentials if we are using OAuth2.0
+    if isinstance(credentials, GoogleAnalyticsCredentialsOAuth):
+        credentials.auth(scope=scope, redirect_uri=redirect_uri)
+        credentials = Credentials.from_authorized_user_info(info={
+            "client_id": credentials.client_id,
+            "client_secret": credentials.client_secret,
+            "refresh_token": credentials.refresh_token,
+            "token": credentials.access_token
+        })
+    # use service account to authenticate if not using OAuth2.0
+    else:
+        credentials = credentials.to_service_account_credentials()
+
+    # Build the service object for Google Analytics api.
+    client = BetaAnalyticsDataClient(credentials=credentials)
+    # get metadata needed for some resources
+    metadata = get_metadata(client=client, property_id=property_id)
+    resource_list = [metadata | metrics_table, metadata | dimensions_table]
+    for query in queries:
+        name = query["resource_name"]
+        dimensions = query["dimensions"]
+        metrics = query["metrics"]
+        resource_list.append(dlt.resource(basic_report(client=client, rows_per_page=rows_per_page, property_id=property_id, dimensions=dimensions, metrics=metrics,
```

Comment:
do we really want to replace reports? if you do them daily I assume you want to keep daily reports. what is a typical use case @adrianbr ? running this daily and keep reports for given day?